input/system/selfhosted/otel_handler.go

```diff
@@ -75,6 +76,10 @@ func skipDueToK8sFilter(kubernetes *common.KeyValueList, server *state.Server, p
 	var k8sPodName string
 	var k8sNamespaceName string
 
+	if kubernetes == nil {
+		return false
+	}
+
```

Comment:
I'd personally keep that away form `skipDueToK8sFilter` which might be missleading, because this function is only about filtering out results and not assessing payload's contents.

Reply:
What do you mean? I'm only guarding against `kubernetes *common.KeyValueList` which could be nil, but the function doesn't seem to expect it to be nil so it panic crashes..

Reply:
Something like

```go
if kubernetes != nil && skipDueToK8sFilter(kubernetes, server, prefixedLogger) {
	continue
}
```

This is just a style comment where to leave the `skipDueToK8sFilter` doing only what it is meant to be. No big deal here.

Reply:
I actually didn't end up including this in my open PR for other OTEL things, but I did originally have this and also prefer failing early here rather than inside of the function for what it's worth.

https://github.com/pganalyze/collector/pull/719/files#diff-9b41a22ed8cdaeb1a5377b0a3245b598b04071bfda5edd565de5707b5b5cef24R170

Reply:
I agree with the idea to move this to where we call the function, I've applied that change in my rebased version in https://github.com/pganalyze/collector/pull/747

Regarding the use of a pointer/reference here - I think its better to keep it as a reference, to avoid a potential unnecessary copy of the struct.